blob/fileblob/fileblob.go

```diff
@@ -600,13 +606,16 @@ func (b *bucket) NewTypedWriter(ctx context.Context, key string, contentType str
 	if len(opts.Metadata) > 0 {
 		metadata = opts.Metadata
 	}
-	attrs := xattrs{
-		CacheControl:       opts.CacheControl,
-		ContentDisposition: opts.ContentDisposition,
-		ContentEncoding:    opts.ContentEncoding,
-		ContentLanguage:    opts.ContentLanguage,
-		ContentType:        contentType,
-		Metadata:           metadata,
+	var attrs *xattrs
```

Comment:
I think instead of using `nil` attrs as a signal that we're not writing the metadata, it might be simpler and more clear to leave this as-is, and instead add a  `SkipMetadata` field to `writer` (initialized via `opts.SkipMetadata`).

Then just the `setAttrs` call and the `os.Remove` call need to check it.